spec/03-plugins/08-datadog/01-log_spec.lua

```diff
@@ -168,4 +182,33 @@ describe("Plugin: datadog (log)", function()
     assert.contains("kong.datadog3_com.request.status.200:1|c|#T2:V2,T3:V3,T4", gauges)
     assert.contains("kong.datadog3_com.latency:%d+|g|#T2:V2:V3,T4", gauges, true)
   end)
+
+  it("should not return a runtime error (regression)", function()
+    local thread = threads.new({
+      function()
+        local socket = require "socket"
+        local server = assert(socket.udp())
+        server:settimeout(1)
+        server:setoption("reuseaddr", true)
+        server:setsockname("127.0.0.1", 9999)
+        local gauge = server:receive()
+        server:close()
+        return gauge
+      end
+    })
+    thread:start()
+
+    local res = assert(client:send {
+      method = "GET",
+      path = "/NonMatch",
+      headers = {
+        ["Host"] = "fakedns.com"
+      }
+    })
+
+    assert.res_status(404, res)
+    
+    local err_log = pl_file.read(helpers.test_conf.nginx_err_logs)
+    assert.is_nil(err_log:find("attempt to index field 'api' (a nil value)", nil, true))
```

Comment:
this doesn't seem to work. is there an equivalent to not_find or not_contains? I am not seeing any documentation for assert.not_matches but I think it is doing something different than find.